### PR TITLE
[IndexTable] Design Updates and assorted bug fixes

### LINKED ISSUE #8354: [IndexTable] Sticky headers not respecting flush = true to remove default padding
## Issue summary

Currently when you have an IndexTable header with flush as true you still get default padding on the sticky header. This then causes alignment problems with the rest of the column if you're expecting there to be no padding.

## Expected behavior

Sticky headers respect the value of `flush` applied to the header

## Actual behavior

Sticky headers appear to always get default padding.

Example with the default headers and the Price header has flush = true
Screenshot 2023-02-14 at 3 05 44 PM

Once sticky headers are visible
Screenshot 2023-02-14 at 3 06 00 PM

## Steps to reproduce the problem

1. Create an IndexTable with a header that has flush = true
2. Scroll to where sticky headers are visible
3. Observe the padding on the header with flush = true

## Reduced test case

This can be observed on the editor pages for the catalogs surface. I can provide a spin instance if needed.

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 10.27.2
- Browser: Google Chrome
- Device: Apple M1 Pro
- Operating System: Ventura 13.2

### LINKED ISSUE #8355: [IndexTable] Sticky headers are misaligned at certain viewport dimensions
## Issue summary

It seems like there may be an issue with sticky header alignment once media queries are applied for smaller viewports.

## Expected behavior

Sticky headers remain aligned with their columns

## Actual behavior

Example with a very small viewport
Screenshot 2023-02-14 at 3 22 15 PM

## Steps to reproduce the problem

1. Go to any IndexTable while having a small window
2. Scroll to where sticky headers are visible
3. Observe the sticky header alignment compared to the column content

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 10.27.2
- Browser: Google Chrome
- Device: Apple M1 Pro
- Operating System: Ventura 13.2

### PR DESCRIPTION
**Pull request by jesstelford**
Fixes https://github.com/Shopify/polaris-summer-editions/issues/478

This PR also includes fixes for the following when design updates are enabled: 
* https://github.com/Shopify/polaris-summer-editions/issues/263
* https://github.com/Shopify/polaris-summer-editions/issues/354
* https://github.com/Shopify/polaris-summer-editions/issues/353
* https://github.com/Shopify/polaris/issues/8355
* https://github.com/Shopify/polaris/issues/8354
* https://github.com/Shopify/polaris-summer-editions/issues/445

[Storybook](https://5d559397bae39100201eedc1-yjmcfkotfo.chromatic.com/?path=/story/all-components-indextable--default&globals=polarisSummerEditions2023:true)

### CODE REVIEW

**Comment by gwyneplaine** [comment]
@sarahill  I've updated the styles for `status= 'subdued` prop to be the same as `subdued`.  Previously these were different, the former having `p-color-text` and the latter using `p-color-text-subdued`. Normalised both to use the same styles for consistency. 

Before: 
Screenshot 2023-06-22 at 11 25 58 am

After: 
Screenshot 2023-06-22 at 11 25 03 am

**Comment by sarahill** [comment]
1. Condensed + Small screen examples: 
Looks like there's an extra top border or something hanging out?
There's also some extra spacing at the bottom but I might just be noticing that because there are only 2 rows assuming that space is always there.
image
image


Nice job! This is the only thing I'm noticing that feels off.

**Comment by gwyneplaine** [comment]
@sarahill while rebasing on top of `main` we discovered @aaronccasanova's [updates](https://github.com/Shopify/polaris/pull/9498/files#diff-e9d0c2a006c030f315115b92e46c3b44667bef0b5e909e1b3af852e6eb2b605b) from yesterday 

This looks to be a very comprehensive and thorough replacement of usages of `--p-color-bg-subdued` to `--p-color-bg-secondary-experimental` as per the conversation in this [issue](https://github.com/Shopify/polaris-summer-editions/issues/262). This seems sensible for table headings in `IndexTable` as per the problem described in the issue linked above, but we've reverted these changes for `subdued` states, as we are **assuming** these are to remain `bg-subdued` as per [Figma](https://www.figma.com/file/jLLkmo9r28hiqPvf4s90E4/Polaris-Uplift-Components-%5Bgen3%E2%80%93alpha%5D?type=design&node-id=87878-72468&mode=design&t=xl5HMYbsdbFdXpeW-4). 

The only point of confusion for us that we'd love clarification on is that the table heading in figma is also using the `bg-subdued` token rather than the `bg-secondary` as per Aaron's changes. Is Figma out of date in this case, or do we need to revert that part of Aaron's change too? 

No stress, just trying to get some clarity 🙏 

As an aside, the difference is very minimal rgba(247,247,247) for `secondary` and rgb(243,243,243) for `subdued`. 
Screenshot 2023-06-23 at 3 04 54 pm

**Comment by gwyneplaine** [comment]
> 1. Condensed + Small screen examples:
>    Looks like there's an extra top border or something hanging out?

This is looks to be an existing pre-uplift bug, but we've also now resolved it for uplift in this PR 👍

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 1]
```diff
@@ -228,7 +248,10 @@ $loading-panel-height: 53px;
       &,
       .TableCell-first,
       .TableCell-first + .TableCell {
-        background-color: var(--p-color-bg-subdued-hover);
+        // deliberately moved it down in the cascade because otherwise the
+        // selected state overrides the hover state and I think we want it the
+        // other way around.
+        background-color: var(--p-color-bg-hover);
```
Comment (sam-b-rose) [inline polaris-react/src/components/IndexTable/IndexTable.scss:254]: I think we'll need to account for the Zebra strip variation here and on line `269` with the `var(--p-color-bg-primary-subdued-hover)` value.

The background seems different from the rest of the cells on hover when selected and when not selected.

https://github.com/Shopify/polaris/assets/11774595/e1fffd4e-fc8a-4095-9673-83b9b511167f

**Review by sam-b-rose — COMMENTED** [review]
Looks awesome, great clean up and design fixes! Most everything looks good, just one comment regarding the zebra strip states on the first non-checkbox cell on hover. 🦓

File: polaris-react/src/components/IndexTable/IndexTable.scss (hunk 1 above)
Comment (jesstelford) [reply polaris-react/src/components/IndexTable/IndexTable.scss:254]: We've punted design updates for Zebra striping: It's not in use in web and there's a desire to deprecate it in the future, so there's no new designs for what colours should be used on zebra when combined with status=subdued/selected/etc.

For illustrative purposes, I've gathered up all the possible states we'd need to cover in a table:

<details>

<summary>Logic matrix</summary>

|                    | [base]                                                                                                                                                                                                                                                                                                                                                                                                                                          | `subdued`                                                                                                                                                                                                                                                                                                                                                                                                                                       | `zebra`                                                                                                                                                                                                                                                                                                                                                                                                                                          | `status="success"`                                                                                                                                                                                                                                                                                                                                                                                                                              | `status="subdued"`                                                                                                                                                                                                                                                                                                                                                                                                                              |
|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| [base]             | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table>  | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |
| `subdued`          | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table>  | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |
| `zebra`            | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table>  | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |
| `status="success"` | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table>  | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |
| `status="subdued"` | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table>  <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |


</details>

You can see by removing Zebra striping, we are able to leave off ~80 variants 🎉

**Review by sarahill — APPROVED** [review]

File: polaris-react/src/components/IndexTable/IndexTable.scss (hunk 1 above)
Comment (sarahill) [reply polaris-react/src/components/IndexTable/IndexTable.scss:254]: ➕ to what Jess said. Zebra striping is being phased out and is no longer used in web so we decided to do the minimal amount of support for this.

Comment (sam-b-rose) [reply polaris-react/src/components/IndexTable/IndexTable.scss:254]: Awesome! That sounds great, I like the simplification 👍 

Thanks for the context @jesstelford and @sarahill

**Review by sam-b-rose — APPROVED** [review]

**Pull request merged**